src/controllers/addRecipeController.ts

```diff
@@ -0,0 +1,84 @@
+import { Request, Response } from "express";
+import getDatabase from "../Services/databaseConnector";
+import {
+  ingredientIdExists,
+  isIdValid,
+  numberLengthIsValid,
+  stringLengthIsValid,
+  userIdExists,
+} from "../Services/validators";
+import { Connection } from "promise-mysql";
+
+export async function addRecipe(req: Request, res: Response) {
+
+  try {
+    const db: Connection = await getDatabase();
+    const userId: number = Number(req.params.userId);
+
+    if(!isIdValid(userId)) {
+      res.status(400).json({
+        message: "Invalid user id",
+      });
+      return
+    }
+
+    if(!(await userIdExists(db, userId))) {
+      res.status(400).json({
+        message: "Invalid user id",
+      });
+      return
+    }
+
+    let validIngredients: number[] = [];
+    for (let ingredient of req.body.ingredients) {
+      if (await ingredientIdExists(db, ingredient)) {
+        validIngredients.push(ingredient)
+      }
```

Comment:
You actually need to check if this ingredient ID is one of your valid ids too

```suggestion
      if (isIdValid(ingredient) && (await ingredientIdExists(db, ingredient))) {
        validIngredients.push(ingredient)
      }
```

Reply:
done